src/test/ui/suggestions/suggest-change-mut.stderr

```diff
@@ -9,6 +9,10 @@ help: consider removing the leading `&`-reference
    |
 LL |         let mut stream_reader = BufReader::new(stream);
    |                                               --
+help: consider introducing a `where` bound, but there might be an alternative better way to express this requirement
+   |
+LL | fn issue_81421<T: Read + Write>(mut stream: T) where &T: std::io::Read {
```

Comment:
Is this syntax even allowed? `where &T: ..` will not compile due to missing lifetime parameters unfortunately

```rust
error[E0637]: `&` without an explicit lifetime name cannot be used here
 --> src/lib.rs:2:23
  |
2 | fn foo<T>(x: T) where &T: std::io::Read { }
  |                       ^ explicit lifetime name needed here

```

Reply:
We (I?) allow for limited "mistakes" in the suggested code as long as the compiler gives you further suggestions that lead you to working code. We can gate this suggestion in particular or/and extend E0637 to suggest where a lifetime can be added (we already have suggestions to extend `foo`'s type param list and to add a `for` lifetime in other places).

Reply:
Filed https://github.com/rust-lang/rust/issues/82312 to track that potential work.